### ISSUE TITLE
add case-insensitive string comparison operations

### ISSUE THREAD

**Issue by shriram** [brownplt/pyret-lang#860]
String equality, <, >, <=, >=…

The "-ci" naming convention of Racket seems fine for names. These would all be functions; there would be no binary operator equivalent (at least until Pyret gets Haskell-style backticks or Racket-style middots…).

The main semantic issue is figuring out what case-insensitivity _means_, especially in light of unicode. @jpolitz's comment:
> Do you care about Unicode normalization for combining marks vs. a glyph with an accent?  What about "ß" == "SS", etc?
Perhaps it's best to wait for someone to figure it out for Unicode, someone else to port that to JavaScript, and us to just inherit that whole mess with pointers to someone else's documentation explaining the rules.

**Comment by ds26gte** [comment]
Capitalizing the single letter ß as the two letters SS is now passé. Unicode has a capital ß (= ẞ) at U+1e9e (for at least eight years now!). This helps maintain the invariant that a string's length shouldn't change on up-casing it (Racket unfortunately still violates this).

**Comment by blerner** [comment]
I don't think that invariant applies to all of Unicode -- and even with esszet, if that letter is accented then then NFC and NFD normalizations might not be the same length.  I don't want to touch this until someone else solves unicode for us.

**Comment by jpolitz** [comment]
We could simply specify one that converts both to `tolower` and compares them that way.  We sort of have to provide `tolower` because it's just so useful.  Then we could write `string-compare-lc` and `string-compare-uc` which are defined to do the lower/upper conversion on _both_ strings and compare them.
